server/src/main/java/org/elasticsearch/index/mapper/DocCountFieldMapper.java

```diff
@@ -115,4 +121,42 @@ protected String contentType() {
     public static IndexableField field(int count) {
         return new CustomTermFreqField(NAME, NAME, count);
     }
+
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        return new SyntheticFieldLoader();
+    }
+
+    private class SyntheticFieldLoader implements SourceLoader.SyntheticFieldLoader {
+        private PostingsEnum postings;
+        private boolean hasValue;
+
+        @Override
+        public Stream<Map.Entry<String, StoredFieldLoader>> storedFieldLoaders() {
+            return Stream.empty();
+        }
+
+        @Override
+        public DocValuesLoader docValuesLoader(LeafReader leafReader, int[] docIdsInLeaf) throws IOException {
+            postings = leafReader.postings(new Term(DocCountFieldMapper.NAME, DocCountFieldMapper.NAME));
```

Comment:
nit: Does it make sense to make `﻿new Term(DocCountFieldMapper.NAME, DocCountFieldMapper.NAME)` a `static final` constant?